utils.ml

```diff
@@ -55,3 +46,49 @@ module TimeHelper = struct
     | `Null -> Ok None
     | _ -> Error (`Msg "invalid json for Ptime.t option")
 end
+
+module Email = struct
+  type config = {
+    smtp_host : string;
+    smtp_port : int;
+    username : string;
+    password : string;
+  }
+
+  let validate_email email =
+    match Emile.of_string (Json.clean_string email) with
+    | Ok _ -> true
+    | Error s ->
+        Logs.err (fun m -> m "Emile-Email-Validation: %a" Emile.pp_error s);
+        false
+
+  let generate_signature uuid timestamp =
+    let timestamp_str = TimeHelper.string_of_ptime timestamp in
+    Base64.encode_string
+      (Cstruct.to_string
+         (Mirage_crypto.Hash.SHA256.digest
+            (Cstruct.of_string (uuid ^ "-" ^ timestamp_str))))
+
+  let generate_verification_link uuid timestamp =
+    let signature = generate_signature uuid timestamp in
+    let encoded_uuid = Base64.encode_string uuid in
+    "/auth/verify/token=" ^ encoded_uuid ^ "/" ^ signature
+
+  let send_verification_email _email _now =
+    let config_data =
+      {
+        smtp_host = "sandbox.smtp.mailtrap.io";
+        smtp_port = 25;
+        username = "2b08693767f1d9";
+        password = "55168d154ddc56";
+      }
+    in
+    let authentication =
+      {
+        Sendmail.mechanism = PLAIN;
+        username = config_data.username;
+        password = config_data.password;
+      }
```

Comment:
this should be part of the configuration -- either passed via command line arguments or stored in the persistent storage...